test/utils/readRowsImpl2.ts

```diff
@@ -0,0 +1,328 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ServerWritableStream} from '@grpc/grpc-js';
+import {protos} from '../../src';
+import {GoogleError, Status} from 'google-gax';
+
+const VALUE_SIZE = 1;
+// we want each row to be splitted into 2 chunks of different sizes
+const CHUNK_SIZE = 1;
+const CHUNK_PER_RESPONSE = 1;
+
+const DEBUG = process.env.BIGTABLE_TEST_DEBUG === 'true';
+
+export function debugLog(text: string) {
+  if (DEBUG) {
+    console.log(text);
+  }
+}
+
+function prettyPrintRequest(
```

Comment:
please add a todo to simplify this. A lot of this complexity can be simplified, but we dont have time to do that now

Reply:
Done

Reply:
once this is merged, an issue should be opened to track it as well

Reply:
Done